lib.php

```diff
@@ -156,6 +158,14 @@ function scormremote_pluginfile($course, $cm, $context, $filearea, $args, $force
         return false;
     }
 
+    // Handle special case mime types.
+    $mimetype = optional_param('mimetype', null, PARAM_RAW);
```

Comment:
I would not make this a query param as it potentially opens security holes

I'd just make it emit the header just before calling send_file or similar

Reply:
It had to be conditional as scormremote_pluginfile serves html files as well.
I could change the condition to check the extension .js and emit the header on that case.
Not sure about security issue, there are exposed params already including the file type (.js extension)